hashicorp/dev-portal
# fix: adjust featured-in-collections layout

## ✅ "Ready for Review" checklist


---

## 🔗 Relevant links

- [Preview link][/vault/tutorials/adp/transform#summary] 🔎
- [Asana task][task] 🎟️

## 🗒️ What

Fixes an issue where `FeaturedInCollections` did not match design specs.

Specifically, when there is one collection card, it should not occupy the full layout width on large viewports, and should instead appear in a 2-column layout.

## 🤷 Why

To match design spec.

## 🛠️ How

- Updates `CardsGridList` to retain its default "min card width" layout mode, while supporting a "fixed columns" mode via a `fixedColumns` prop
    - This is already used for ordered collections (`fixedColumns={1}`)
- Replaces the one-off card grid implementation for `FeaturedInCollections` with `CardsGridList`, using `fixedColumns={2}` when there is exactly `1` other collection card to render.

## 📸 Design Screenshots

### Before

<img width="1384" alt="CleanShot 2022-05-31 at 12 40 40@2x" src="https://user-images.githubusercontent.com/4624598/171228109-536c9a39-9bc2-405b-b681-8e5ce3544cb4.png">

### After

<img width="1384" alt="CleanShot 2022-05-31 at 12 41 27@2x" src="https://user-images.githubusercontent.com/4624598/171228227-954ac13e-bef4-4789-a3d8-518849128aaa.png">

## 🧪 Testing

- [ ] Visit a tutorial link with one other collection context, such as [/vault/tutorials/adp/transform][/vault/tutorials/adp/transform#summary]
    - Ensure the single collection card at the bottom of the page occupies only one of two columns
- [ ] Visit a tutorial link with many other collection contexts, such as [/waypoint/tutorials/get-started-docker/get-started-install][/waypoint/tutorials/get-started-docker/get-started-install]
    - Ensure the collection cards still appear as expected
- [ ] Visit an ordered collection page, such as [/waypoint/tutorials/get-started-docker][/waypoint/tutorials/get-started-docker]
    - Ensure the cards still appear as expected, in one column only
- [ ] Visit a page with other uses of `CardsGridList`, such as [/waypoint][/waypoint]
    - Cards should appear identical to before this PR; the new layout prop option in `CardsGridList` should not affect existing uses.

[task]: https://app.asana.com/0/1202097197789424/1202341387965022/f
[/vault/tutorials/adp/transform#summary]: https://dev-portal-git-zscards-grid-list-fixed-columns-hashicorp.vercel.app/vault/tutorials/adp/transform#summary
[/waypoint/tutorials/get-started-docker/get-started-install]: https://dev-portal-git-zscards-grid-list-fixed-columns-hashicorp.vercel.app/waypoint/tutorials/get-started-docker/get-started-install
[/waypoint/tutorials/get-started-docker]: https://dev-portal-git-zscards-grid-list-fixed-columns-hashicorp.vercel.app/waypoint/tutorials/get-started-docker
[/waypoint]: https://dev-portal-git-zscards-grid-list-fixed-columns-hashicorp.vercel.app/waypoint
